tsl/test/isolation/specs/deadlock_recompress_chunk.spec.in

```diff
@@ -0,0 +1,99 @@
+# This file and its contents are licensed under the Timescale License.
+# Please see the included NOTICE for copyright information and
+# LICENSE-TIMESCALE for a copy of the license.
+
+# Test that concurrent queries and recompress_chunk cannot
+# deadlock.
+#
+# This can occur if recompressing since a recompress will first do a
+# decompression of the compressed chunk into the uncompressed chunk.
+#
+# We should only have a single chunk here, otherwise we cannot lock
+# the right chunk.
+#
+# We have fetched the actual recompress procedure from the policy and
+# re-implement it here to be able to test that it works as
+# expected. Please check policy_internal.sql to see the code it is
+# based on.
+setup {
+    CREATE OR REPLACE FUNCTION debug_waitpoint_enable(TEXT)
+    RETURNS VOID LANGUAGE C VOLATILE STRICT
+    AS '@TS_MODULE_PATHNAME@', 'ts_debug_point_enable';
+
+    CREATE OR REPLACE FUNCTION debug_waitpoint_release(TEXT)
+    RETURNS VOID LANGUAGE C VOLATILE STRICT
+    AS '@TS_MODULE_PATHNAME@', 'ts_debug_point_release';
+
+    CREATE OR REPLACE FUNCTION debug_waitpoint_id(TEXT)
+    RETURNS BIGINT LANGUAGE C VOLATILE STRICT
+    AS '@TS_MODULE_PATHNAME@', 'ts_debug_point_id';
+
+    CREATE OR REPLACE PROCEDURE recompress_chunk_procedure(regclass) AS $$
+    BEGIN
+      PERFORM decompress_chunk($1, if_compressed => true);
+      PERFORM pg_advisory_lock(debug_waitpoint_id('recompress_after_decompress'));
+      PERFORM pg_advisory_unlock(debug_waitpoint_id('recompress_after_decompress'));
```

Comment:
I don't think those operations do anything. These waitpoints have not been defined in the C code. Shouldn't `debug_waitpoint_id()` throw an exception here?